packages/@headlessui-react/src/hooks/document-overflow/adjust-scrollbar-padding.ts

```diff
@@ -0,0 +1,32 @@
+import { ScrollLockRequest } from './request'
+
+export function adjustScrollbarPadding(
+  req: ScrollLockRequest,
+  next: (req: ScrollLockRequest) => void
+) {
+  let { doc } = req
+
+  if (!req.isLocked) {
```

Comment:
While I get the concept of locks. I wonder if we can use a more intuitive name, or if we can split up the set/(unset/cleanup) things. Or maybe even lock/unlock, like:
```js
return {
  lock() { /* do something */ },
  unlock() { /* do the inverse */ }
}
```
Or a callback/cleanup function like
```js
function apply() {
  // do the thing
  return () => {
    // do the inverse
  }
}
```
Which means that in some scenario's this could become:
```js
function apply() {
  return style(el, 'foo', '12px')
}
```

But this might become a bit harder if you want the concept of a `before`, `after` or `both` middleware (depending on when and where you call `next(req)`).

Might be a bad idea though not sure. Just a bit confusing what "locked" means in this scenario. Like is the document itself in an "unlocked" state and we need to "lock" it by applying the `overflow: hidden`, `padding: scrollbarwidth`, ...?

Reply:
The idea was more like isLocked === scrolling is locked into place but yeah I'll see if I can come up with a better name for this.

Reply:
And yeah I considered doing a non-middleware approach to this but it seemed fragile-ish because of the order I need code to run in. I'll keep investigating though. Maybe there's another API that'll work here. 🤔

Reply:
I had a bit of an aha moment around the API. I've applied it to the React version only for now. If you could give that a look and let me know what you think about it that would be super helpful.